### ISSUE TITLE
How to specify personal-addresses without recreating the db?

### ISSUE THREAD

**Issue by tmalsburg** [djcb/mu#1631]
Just updated mu4e after a longer time.  Reply all now includes my own address in CC.  Apparently that happens because `mu4e-personal-addresses` returns `nil`.  I see that I can tell mu server about my addresses during init, but that means I have to rebuild my db which takes a long time.  Is there a better way to inform mu server about my addresses without rebuilding the db?  (Why does the server need to know my address anyway?  And why is mu4e not using my address as configured in Emacs?  It's surprising that mu4e pretends not to know my address when I configured it in Emacs.)

## Versions of mu, mu4e/emacs, operating system etc.

Current development Mu/Mu4e and Emacs (master branch).

p.s: Sorry if this is explained somewhere. I looked but couldn't find it.

**Comment by djcb** [comment]
The mu server needs to know the address(es) when scanning your messages, so it can determine what are 'personal' addresses (and 'personal messages', but that's something for the future).

People sometimes ran `mu index` with the necessary `--my-address` parameters, which gave confusing results when later on searching for 'personal addresses'. Furthermore, people didn't have the same addresses in their mu4e and mu-by-hand setup. Both are indeed _easy_ to forget. 

So, the solution for this is to specify the addresses only once (with `mu init`), and have those automatically used by `mu4e` as well. Makes it easier for users, who get consistent behavior, and for the mu/mu4e maintainers.

We try to update the the database automatically, but in some cases, indeed, you need to run `mu init`, and then a full `mu index`; this takes a while (a few minutes?), but only needs to be done once. That seems a price worth paying... or?

That said, perhaps we could issue a warning if your emacs-specified email address is not part of the e-mail addresses specified for mu.

**Comment by tmalsburg** [comment]
This all makes sense; thanks for explaining.

The trouble is that it's not clear to the user that the way the index was set up in mu (one program) determines how reply-to-all works in Emacs (another program).  I had to look at several source files and the man page of `mu init` to figure this out.  It's also not clear why `user-mail-address` is being ignored when determining whether the sender should be added to CC.

This could be solved either by issuing a warning (as you say) or by using the addresses from mu plus the value of `user-mail-address`, or, perhaps even better, by doing both.

It might also be good to have warnings for users upgrading from older mu who haven't defined personal addresses in the index (`mu init` doesn't complain when personal addresses are omitted) and for those who still define `mu4e-user-mail-address-list` in their Emacs configuration.

**Comment by djcb** [comment]
Yeah, good idea, I have done so now.
